src/python/pants/core/util_rules/adhoc_process_support.py

```diff
@@ -68,32 +59,48 @@ class AdhocProcessResult:
     adjusted_digest: Digest
 
 
-@rule_helper
-async def _execution_environment_from_dependencies(adhoc_process_target: Target) -> Digest:
+@dataclass(frozen=True)
+class ResolveExecutionDependenciesRequest:
+    address: Address
+    execution_dependencies: tuple[str, ...] | None
+    dependencies: tuple[str, ...] | None  # can go away after 2.17.0.dev0 per deprecation
```

Comment:
Should these be the fields instead of the field values?

That would allow using `request.execution_dependencies.alias` in the `_descr` string below.

Reply:
My goal here was to make `adhoc_process_support` _not_ depend on any custom field types. Happy to discuss the stylistic merits of it, but you're right, not being able to use `.alias` is a downside here.

Reply:
(this is another request type that will go away when we support scoped dependencies; sigh :))